src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc

```diff
@@ -1365,36 +1365,48 @@ ChannelArgs BuildBalancerChannelArgs(
-      // Just as for \a GRPC_ARG_DEFAULT_AUTHORITY, the LB channel should be
-      // treated as a stand-alone channel and not inherit this argument from the
-      // args of the parent channel.
-      .Remove(GRPC_SSL_TARGET_NAME_OVERRIDE_ARG)
-      // Don't want to pass down channelz node from parent; the balancer
-      // channel will get its own.
-      .Remove(GRPC_ARG_CHANNELZ_CHANNEL_NODE)
-      // Remove the channel args for channel credentials and replace it
-      // with a version that does not contain call credentials. The loadbalancer
-      // is not necessarily trusted to handle bearer token credentials.
-      .Remove(GRPC_ARG_CHANNEL_CREDENTIALS)
+
+  ChannelArgs grpclb_channel_args;
+  if (args.Contains(GRPC_ARG_GRPCLB_CHANNEL_ARGS)) {
+    grpclb_channel_args = ChannelArgs::FromC(
+        args.GetPointer<grpc_channel_args>(GRPC_ARG_GRPCLB_CHANNEL_ARGS));
+  } else {
+    // Set grpclb_channel_args based on the parent channel's channel args.
+    grpclb_channel_args =
+        args
+            // LB policy name, since we want to use the default (pick_first) in
+            // the LB channel.
+            .Remove(GRPC_ARG_LB_POLICY_NAME)
+            // Strip out the service config, since we don't want the LB policy
+            // config specified for the parent channel to affect the LB channel.
+            .Remove(GRPC_ARG_SERVICE_CONFIG)
+            // The channel arg for the server URI, since that will be different
+            // for the LB channel than for the parent channel.  The client
+            // channel factory will re-add this arg with the right value.
+            .Remove(GRPC_ARG_SERVER_URI)
+            // The fake resolver response generator, because we are replacing it
+            // with the one from the grpclb policy, used to propagate updates to
+            // the LB channel.
+            .Remove(GRPC_ARG_FAKE_RESOLVER_RESPONSE_GENERATOR)
+            // The LB channel should use the authority indicated by the target
+            // authority table (see \a ModifyGrpclbBalancerChannelArgs),
+            // as opposed to the authority from the parent channel.
+            .Remove(GRPC_ARG_DEFAULT_AUTHORITY)
+            // Just as for \a GRPC_ARG_DEFAULT_AUTHORITY, the LB channel should
+            // be treated as a stand-alone channel and not inherit this argument
+            // from the args of the parent channel.
+            .Remove(GRPC_SSL_TARGET_NAME_OVERRIDE_ARG)
+            // Don't want to pass down channelz node from parent; the balancer
+            // channel will get its own.
+            .Remove(GRPC_ARG_CHANNELZ_CHANNEL_NODE)
+            // Remove the channel args for channel credentials and replace it
+            // with a version that does not contain call credentials. The
+            // loadbalancer is not necessarily trusted to handle bearer token
+            // credentials.
+            .Remove(GRPC_ARG_CHANNEL_CREDENTIALS);
+  }
+
+  return grpclb_channel_args
```

Comment:
Suggest moving the code from lines 1361-1367 down here, since this is where we're using the objects created there.

Reply:
Done.